server/src/main/java/org/opensearch/search/aggregations/bucket/terms/NumericTermsAggregator.java

```diff
@@ -145,6 +171,93 @@ public void collect(int doc, long owningBucketOrd) throws IOException {
         });
     }
 
+    public StarTreeBucketCollector getStarTreeBucketCollector(
+        LeafReaderContext ctx,
+        CompositeIndexFieldInfo starTree,
+        StarTreeBucketCollector parent
+    ) throws IOException {
+        assert parent == null;
+        StarTreeValues starTreeValues = StarTreeQueryHelper.getStarTreeValues(ctx, starTree);
+        return new StarTreeBucketCollector(
+            starTreeValues,
+            StarTreeTraversalUtil.getStarTreeResult(
+                starTreeValues,
+                StarTreeQueryHelper.mergeDimensionFilterIfNotExists(
+                    context.getQueryShardContext().getStarTreeQueryContext().getBaseQueryStarTreeFilter(),
+                    fieldName,
+                    List.of(DimensionFilter.MATCH_ALL_DEFAULT)
+                ),
+                context
+            )
+        ) {
+            @Override
+            public void setSubCollectors() throws IOException {
+                for (Aggregator aggregator : subAggregators) {
+                    this.subCollectors.add(((StarTreePreComputeCollector) aggregator).getStarTreeBucketCollector(ctx, starTree, this));
+                }
+            }
+
+            SortedNumericStarTreeValuesIterator valuesIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
+                .getDimensionValuesIterator(fieldName);
+
+            String metricName = StarTreeUtils.fullyQualifiedFieldNameForStarTreeMetricsDocValues(
+                starTree.getField(),
+                "_doc_count",
+                MetricStat.DOC_COUNT.getTypeName()
+            );
+            SortedNumericStarTreeValuesIterator docCountsIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
+                .getMetricValuesIterator(metricName);
+
+            @Override
+            public void collectStarTreeEntry(int starTreeEntry, long owningBucketOrd) throws IOException {
+                if (valuesIterator.advance(starTreeEntry) == NO_MORE_DOCS) {
+                    return;
+                }
+                long dimensionValue = valuesIterator.nextValue();
+                // Only numeric & floating points are supported as of now in star-tree
+                // TODO: Add support for isBigInteger() when it gets supported in star-tree
+                if (valuesSource.isFloatingPoint()) {
+                    double doubleValue = ((NumberFieldMapper.NumberFieldType) context.mapperService().fieldType(fieldName)).toDoubleValue(
+                        dimensionValue
+                    );
+                    dimensionValue = NumericUtils.doubleToSortableLong(doubleValue);
```

Comment:
We can revisit this later post 2.19 / probably create an issue - as we are doing 2 conversions unnecessarily.

For example if field type is float :  
1. float to double
2. double to long 
3. long to double
to fit into the existing interfaces.

[ currently as well , we're doing SortedNumericDoubleValues -> fieldData.toSortableLongBitseInts ->  NumericUtils.sortableLongToDouble for the original segment doc values , so okay to take it up post 2.19 ]

Reply:
Sure - I have noted post 2.19 items in [Search Plan](https://github.com/opensearch-project/OpenSearch/issues/15257#issuecomment-2621191489) to circle back after this PR.

Reply:
@msfroh @sandeshkr419 is there any optimizations that you can think of here ?

Reply:
As a follow-up optimization, we can think of caching the fieldType - to avoid some of the data type checks. That will most likely be a minor optimization.

Alternatively, can also think of setting up a new CollectionStrategy to avoid this multiple conversions and handle the data type conversion only once during collectBucket phase.

Certainly, I think we should handle this experiment outside of this PR as I think we may have a chance to optimize the default non-star tree aggregation as well here. Thoughts @bharath-techie @msfroh?

Reply:
Makes sense as there is scope of optimization for default and star tree aggregators.